include/exadg/operators/operator_base.cpp

```diff
@@ -822,26 +813,28 @@ void
 OperatorBase<dim, Number, n_components>::internal_calculate_system_matrix(
   SparseMatrix & system_matrix) const
 {
-  // assemble matrix locally on each process
-  if(evaluate_face_integrals() and is_dg)
+  if(is_dg and evaluate_face_integrals())
   {
-    matrix_free->loop(&This::cell_loop_calculate_system_matrix,
-                      &This::face_loop_calculate_system_matrix,
-                      &This::boundary_face_loop_calculate_system_matrix,
-                      this,
-                      system_matrix,
-                      system_matrix);
+    dealii::MatrixFreeTools::compute_matrix(*matrix_free,
+                                            matrix_free->get_affine_constraints(),
+                                            system_matrix,
+                                            &This::cell_compute_diagonal_matrix,
+                                            &This::face_compute_diagonal_matrix,
+                                            &This::boundary_face_compute_diagonal_matrix,
```

Comment:
Here we want to compute the full matrix, not just the diagonal. So I assume this is a bug here?

Reply:
Ah, I think I am just irritated by the name. Nevertheless, it seems as if the code does not work correctly and I get a matrix with `n_nonzero_elements() = 0`.